README.md

```diff
@@ -1,62 +1,144 @@
+### External reference
+A D:// transaction with external reference is formatted like:
 
 ```
 OP_RETURN
-  19iG3WTYSsbyos3uJ733yK4zEioi1FesNU
-  [alias]
+  19iG3WTYSsbyos3uJ733yK4zEioi1FesNu
+  [key]
   [pointer]
   [type]
   [sequence]
 ```
 
-*  alias: name,filename,alias
-*  pointer: txid of b:// or c://
-*  type: 'c' or 'b'
-*  sequence: only needed if multiple tx(update) in same block. sequence is a number, higher number tx overwrites the lower one. everything what is not number is translated to 1.
+*  `key`: a utf8 encoded string no longer than 1024 chars __not__ starting with `/` and not including the control chars `[\x00-\x1F\x7F]`. It is suggested to simulate a folder like structure. Even if all utf8 chars are allowed it is advised to consider, that normal day usage will involve a URL escaped version presented to the user. Only using `[a-zA-Z0-9_~/@!$&*+,.:;=-]` is therefore advisable.
 
+*  `pointer`: string with txid of b:// or hash of a c://
 
-#### Overwrite D:// State
-New transactions with the same alias from a sender automatically override the old state. Protocol API always outputs only the most current state.
-The alias sender combination prevents an unauthorized one from changing state.
+*  `type`: The string `c` or `b` indicating the type of pointer
+
+*  `sequence`: Integer larger than 0 and smaller than 2^53-1. Everything that is not a number or a negative number is considered to be `1`. Only used to indicate the order of events if multiple updates are provided in the same block to the same key from the same owner. 
+
+
+
+
+
+### Internal reference
+A D:// transaction with internal reference is piped directly on to a B:// formatted structure with mandatory fields for encoding and filename:
+
+```
+OP_RETURN
+  19HxigV4QyBv3tHpQVcUEQyq1pzZVdoAut
+  [DATA]
+  [MEDIA TYPE]
+  [ENCODING] or NULL
+  [FILENAME] or NULL
+  |
+  19iG3WTYSsbyos3uJ733yK4zEioi1FesNu
+  [key]
+  NULL
+  NULL
+  [sequence]
+```
+
+
+
+
+
+
+### Explicit owner
+
+Both transactions with internal and external references can have the owner explicitly provided by piping the AIP protocol after the demonstrated fields (see https://github.com/BitcoinFiles/AUTHOR_IDENTITY_PROTOCOL). 
+
+
+```
+OP_RETURN
+  19iG3WTYSsbyos3uJ733yK4zEioi1FesNu
+  [key]
+  [pointer]
+  [type]
+  [sequence]
+  |
+  15PciHG22SNLQJXMoSUaWVi7WSqc7hCfva
+  [Signing Algorithm]
+  [Signing Address]
+  [Signature]
+```
+
+
+
+```
+OP_RETURN
+  19HxigV4QyBv3tHpQVcUEQyq1pzZVdoAut
+  [DATA]
+  [MEDIA TYPE]
+  [ENCODING] or NULL
+  [FILENAME] or NULL
+  |
+  19iG3WTYSsbyos3uJ733yK4zEioi1FesNu
+  [key]
+  NULL
+  NULL
+  [sequence]
+  |
+  15PciHG22SNLQJXMoSUaWVi7WSqc7hCfva
+  [Signing Algorithm]
+  [Signing Address]
+  [Signature]
+```
+
+If a transaction includes a valid AIP signature and the signature involves all previous fields (AIP_ALL) it is not the sender but the signing address that will be considered the owner in the transaction. 
```

Comment:
Something to think about.... what if someone makes a D transaction with AIP and then later makes other D transactions without AIP (but using the same address via funding vin address?)

Should it be simultaneously acceptable to consider the funding address and the AIP as "valid anchors" to the content?

Instead of thinking about as "ownership" we think of the address as a "reference".

Reply:
I think all ways should be supported. If D comes without AIP the vin address is taken and if AIP is present this is ignored.